tests/unit/test-util/test.submit-addon.js

```diff
@@ -80,6 +81,7 @@ describe('util.submit-addon', () => {
       const apiHost = 'https://foo.host';
       const downloadDir = '/foo';
       const clientSpy = sinon.spy(Client);
+      const apiAuthSpy = sinon.spy(JwtApiAuth);
```

Comment:
While we are here, let's also assert explicitly that the expected authorization header ends up into the actual request, e.g. something like the following (diff on top of this version of the pull request) may be enough for getting some explicit covered about that:

```diff
diff --git a/tests/unit/test-util/test.submit-addon.js b/tests/unit/test-util/test.submit-addon.js
index fa17b03..5e99197 100644
--- a/tests/unit/test-util/test.submit-addon.js
+++ b/tests/unit/test-util/test.submit-addon.js
@@ -199,19 +199,30 @@ describe('util.submit-addon', () => {
       status: 'unreviewed',
     };
 
+    const getAuthHeaderSpy = sinon.spy(apiAuth, 'getAuthHeader');
+
+    afterEach(() => {
+      getAuthHeaderSpy.resetHistory();
+    });
+
     describe('doUploadSubmit', () => {
       it('submits the xpi', async () => {
         const client = new Client(clientDefaults);
         sinon.stub(client, 'fileFromSync')
           .returns(new File([], 'foo.xpi'));
+
+        const nodeFetchStub = sinon.stub(client, 'nodeFetch');
         mockNodeFetch(
-          sinon.stub(client, 'nodeFetch'),
+          nodeFetchStub,
           `${apiHostPath}/addons/upload/`,
           'POST',
           [
             {
               body: sampleUploadDetail,
               status: 200,
+              headers: {
+                Authorization: 'test123',
+              },
             },
           ]
         );
@@ -224,6 +235,20 @@ describe('util.submit-addon', () => {
         const returnUuid = await client.doUploadSubmit(xpiPath, channel);
         assert.equal(sampleUploadDetail.uuid, returnUuid);
         sinon.assert.calledWith(waitStub, sampleUploadDetail.uuid);
+
+        // Verify the jwt Authorization header are included in the response.
+        sinon.assert.calledOnce(getAuthHeaderSpy);
+        const authHeaderValue = await getAuthHeaderSpy.getCall(0).returnValue;
+        assert.match(authHeaderValue, /^JWT .*/);
+        sinon.assert.calledOnceWithMatch(
+          nodeFetchStub,
+          sinon.match.instanceOf(URL),
+          sinon.match({
+            headers: {
+              Authorization: authHeaderValue,
+            },
+          })
+        );
       });
     });
```

Reply:
re:
```diff
+              headers: {
+                Authorization: 'test123',
+              },
```
AMO don't send the Authorization header in responses.

Reply:
eh, yeah sorry about that, that was a leftover from an initial attempt 
(I was looking into how to pass to the helper the request headers to expect and initially I looked into passing it through the array of the responses for the mocked calls, then I thought that was not looking great and opted for the approach in the rest of the diff).

Thanks for spotting that left over and omit it from the changes applied to this PR.